### PR TITLE
Feature/type inference for push function

### PR DESCRIPTION
**Pull request by JakobSchaefer**
Hi @ibc,

it would be great if you could look over this change.

I have slightly changed the type declarations, so that typescript is able to determine the return value when using the AwaitQueue.push function.

Best regards

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export class AwaitQueue
 		this.pendingTasks.length = 0;
 	}
 
-	dump(): { task: AwaitQueueTask;	name?: string; stopped: boolean }[]
+	dump(): { task: AwaitQueueTask<unknown>;	name?: string; stopped: boolean }[]
```
Comment (ibc) [inline src/index.ts]: Wrong format here, too many spaces.

**Review by ibc — CHANGES_REQUESTED** [review]
Awesome. Please fix the format error an run "npm run lint" to ensure it passes the linting checks.

File: src/index.ts (hunk 1 above)
Comment (JakobSchaefer) [reply src/index.ts]: I don't know how that happened. Should fit now

**Review by ibc — APPROVED** [review]

**Comment by ibc** [comment]
THANKS A LOT. Published in 2.3.0.

**Pull request merged**